docs/core-concepts/networking.md

```diff
@@ -0,0 +1,177 @@
+The Public network address can be used to access network services exposed by the Node.
+
+Each Grid has a single **Overlay Network** using private RFC1918 address space.
+The Kontena Agent establishes the Overlay Network mesh between the Nodes.
+Nodes and Containers are connected to the overlay network.
+Nodes and Containers can use their *overlay* network address to communicate within the Grid.
+
+Each host Node runs the Kontena **Agent**, which establishes a WebSocket connection to a **Master** node running the Kontena **Server**.
+The host Node's Kontena Agent and the Master node are only used for cluster management. The Master node is not connected to the Grid, and Containers attached to the Grid cannot communicate with the Kontena Master.
+The Master Node does not expose any network services other than the HTTP API used by the CLI and Agent.
+
+While a Kontena master can manage multiple Grids, each Grid is an isolated overlay network with its own address space.
+Nodes and Containers attached to different Grids cannot communicate.
+
+## Grid
+
+Each grid has a single overlay network. The default overlay network used is `10.81.0.0/19`, and it currently cannot be configured at creation time nor changed later.
+
+Each Grid may also include a set of Trusted Subnets, which are used for the Overlay Networking described further below.
+
+### IP Address Management
+
+Overlay network addresses are allocated by the Kontena Master, which tracks each allocated overlay network address.
+Using the default address allocation scheme, each Grid can contain up to 254 Host Nodes, and 8190 Service Containers.
+
+When a Host Node's Kontena Agent is disconnected from the Kontena Master, the overlay network addresses used by the Node's Containers are released.
+Released overlay network addresses may be used for new Service Containers being deployed.
+
+If a disconnected host Node reconnects, the overlay network addresses of any returning Service containers are restored.
+In case of a conflict where the overlay network address has been reallocated for a second Service Container after the original Node disconnected, the reallocated overlay network address is released, and the second Service Container is re-deployed.
+
+## Host Node
+
+A host Node is a (virtual) machine running the Docker Engine and Kontena Agent.
+The Kontena Agent runs as a Docker container, and controls the Docker Engine to manage infrastructure Containers and Service Containers.
+
+### Node Network Addresses
+Each host Node has a total of four different network addresses:
+
+* The external Internet address of the machine (`public_ip`)
+
+  The public network address is resolved at startup using the `http://whatismyip.akamai.com` service, or it can be configured using `KONTENA_PUBLIC_IP`.
+
+  The Node's public addess can be used to connect to network services exposed on that host Node, including the Weave service used for the overlay network.
```

Comment:
Clarified this to explicitly talk about "Weave data and control plane connections"